Pal/src/host/Linux-SGX/db_semaphore.c

```diff
@@ -54,14 +54,15 @@ static inline int atomic_dec_if_positive (struct atomic_int *v)
 }
 
 int
-_DkSemaphoreCreate (PAL_HANDLE handle, int initialCount, int maxCount)
+_DkSemaphoreCreate (PAL_HANDLE * sem, int initialCount, int maxCount)
 {
     /*
      * 1. Allocate memory for db_sem (this includes a futex variable).
      * 2. Pack it into a PAL_HANDLE
      * 3. Set the semaphore object with the argument values (count, maxCount)
      */
 
+    PAL_HANDLE handle = malloc(HANDLE_SIZE(semaphore));
```

Comment:
I believe the corresponding free() is in DkObjectClose?  If so, it might be good to at least have a comment here, or perhaps hoist the malloc into common PAL code so that it is obvious that every malloc has a paired free.